addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/XMLConnection.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.yamahareceiver.internal.protocol.xml;
+
+import java.io.BufferedReader;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+import org.openhab.binding.yamahareceiver.internal.protocol.AbstractConnection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * All other protocol classes in this directory use this class for communication. An object
+ * of HttpXMLSendReceive is always bound to a specific host.
+ *
+ * @author David Graeff - Initial contribution
+ *
+ */
+public class XMLConnection extends AbstractConnection {
+    private Logger logger = LoggerFactory.getLogger(XMLConnection.class);
+
+    public final static String XML_GET = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"GET\">";
+    public final static String XML_PUT = "<?xml version=\"1.0\" encoding=\"utf-8\"?><YAMAHA_AV cmd=\"PUT\">";
+    public final static String XML_END = "</YAMAHA_AV>";
+
+    public XMLConnection(String host) {
+        super(host);
+    }
+
+    /**
+     * Post the given xml message
+     *
+     * @param message XML formatted message excluding < ?xml > or <YAMAHA_AV> tags.
+     * @throws IOException
+     */
+    @Override
+    public void send(String message) throws IOException {
+        HttpURLConnection connection = null;
+        if (message.startsWith("<?xml")) {
+            throw new IOException("No preformatted xml allowed!");
+        }
+
+        message = XML_PUT + message + XML_END;
+
+        writeTraceFile(message);
+
+        try {
+            URL url = new URL("http://" + host + "/YamahaRemoteControl/ctrl");
+            connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("POST");
+            connection.setRequestProperty("Content-Length", "" + Integer.toString(message.length()));
+
+            connection.setUseCaches(false);
+            connection.setDoInput(true);
+            connection.setDoOutput(true);
+
+            // Send request
+            DataOutputStream wr = new DataOutputStream(connection.getOutputStream());
```

Comment:
@marcelrv I don't like the HttpUtil class to be honest. It creates one static instance of a HttpClient instance and starts it, but it is never stopped. It should have been modeled as an OSGi service, that on deactivate stops the instance again.